src/Handlers/SendRequestHandler.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { SendRequestHandler } from './SendRequestHandler';
+import { Protocol, Request, RequestExecutor, Response } from '../RequestExecutor';
+import chai, { should } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { anything, deepEqual, instance, mock, verify, when } from 'ts-mockito';
+
+chai.use(chaiAsPromised);
+should();
+
+
+describe('SendRequestHandler', () => {
+  const simpleEvent = {
+    protocol: Protocol.HTTP,
+    method: 'GET',
+    url: 'http://test.con',
+    headers: {},
+    body: 'test'
+  };
+
+  const httpExecutorMock = mock<RequestExecutor>();
+  when(httpExecutorMock.protocol).thenReturn(Protocol.HTTP);
+  const wsExecutorMock = mock<RequestExecutor>();
+  when(wsExecutorMock.protocol).thenReturn(Protocol.WS);
+
+
+  let requestExecutors: RequestExecutor[];
+  let sut: SendRequestHandler;
+  beforeEach(() => {
+    requestExecutors = [];
+    sut = new SendRequestHandler(requestExecutors);
```

Comment:
Use IoC container to manage deps (see [tsyringe](https://github.com/microsoft/tsyringe))